internal/impl/sftp/integration_test.go

```diff
@@ -129,6 +109,133 @@ cache_resources:
 	})
 }
 
+func TestIntegrationSFTPDeleteOnFinish(t *testing.T) {
+	integration.CheckSkip(t)
+	t.Parallel()
+
+	resource := setupDockerPool(t)
+
+	client, err := getClient(resource)
+	require.NoError(t, err)
+
+	writeSFTPFile(t, client, "/upload/1.txt", "data-1")
+	writeSFTPFile(t, client, "/upload/2.txt", "data-2")
+	writeSFTPFile(t, client, "/upload/3.txt", "data-3")
+
+	config := `
+output:
+  drop: {}
+
+input:
+  sftp:
+    address: localhost:$PORT
+    paths:
+      - /upload/*.txt
+    credentials:
+      username: foo
+      password: pass
+    delete_on_finish: true
+    watcher:
+      enabled: true
+      poll_interval: 100ms
+      cache: files_memory
+
+cache_resources:
+  - label: files_memory
+    memory:
+      default_ttl: 900s
+`
+	config = strings.NewReplacer(
+		"$PORT", resource.GetPort("22/tcp"),
+	).Replace(config)
+
+	var receivedPathsMut sync.Mutex
+	var receivedPaths []string
+
+	builder := service.NewStreamBuilder()
+	require.NoError(t, builder.SetYAML(config))
```

Comment:
@rockwotj 
> I don't think there is a utility so either you need to do option 1 or implement the retry logic - which I don't think should be too bad?
> 
> Here's the code that drives this in benthos AFAIK: https://github.com/redpanda-data/benthos/blob/dad70374cd8fb323f0c7f47452498ea94c2ed7aa/internal/component/input/async_reader.go#L115
> 
> The pipeline option (number 1) might be the best route, but I'm not too familiar with that test helper myself.

Knowledge from the great and powerful @mihaitodor

Reply:
Great tests thanks :)